presto-main/src/test/java/io/prestosql/sql/query/TestSubqueries.java

```diff
@@ -96,10 +96,34 @@ public void testCorrelatedSubqueriesWithLimit()
         assertions.assertQuery(
                 "SELECT (SELECT t.a FROM (VALUES 1, 2) t(a) WHERE t.a=t2.b LIMIT 2) FROM (VALUES 1) t2(b)",
                 "VALUES 1");
-        // cannot enforce LIMIT on correlated subquery
+        assertions.assertQuery(
+                "SELECT (SELECT t.a FROM (VALUES 1, 2, 3) t(a) WHERE t.a = t2.b LIMIT 2) FROM (VALUES 1) t2(b)",
+                "VALUES 1");
+        assertions.assertFails(
+                "SELECT (SELECT t.a FROM (VALUES 1, 1, 2, 3) t(a) WHERE t.a = t2.b LIMIT 2) FROM (VALUES 1) t2(b)",
+                "Scalar sub-query has returned multiple rows");
```

Comment:
This exception comes from the execution. So this is planned and optimized, then executed and afterwards we find that more than one row returns so this fails.